libs/fs.js

```diff
@@ -521,13 +521,15 @@ var fs = (function() {
       from = buffer.contentSize;
     }
 
-    var newLength = (from + data.byteLength > buffer.contentSize) ? (from + data.byteLength) : (buffer.contentSize);
+    var newLength = (from + length > buffer.contentSize) ? (from + length) : (buffer.contentSize);
 
     buffer.setSize(newLength);
 
-    buffer.array.set(data, from);
+    for (var i = 0; i < length; i++) {
```

Comment:
Do we have any numbers on this? Also, what is considered small? The time spent in actual copying seems to be very small compared to the other things going during install run. I'd prefer simpler code that doesn't allocate and performs generally well.